### PR TITLE
added DataLoadingConfig to K8sPod message

### PR DESCRIPTION
**Pull request by hamersaw**
# TL;DR
This PR adds the `DataLoadingConfig` information to the `K8sPod` proto message so that Flyte can inject [copilot](https://github.com/flyteorg/flytecopilot) as a sidecar in Pod tasks. This allows the use of `PodTemplate` configuration with `ContainerTasks`. 

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
^^^

## Tracking Issue
https://github.com/flyteorg/flyte/issues/3123

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by flixr** [comment]
@hamersaw anything still needed here until it can be merged?

**Comment by hamersaw** [comment]
@flixr we typically don't merge until we have e2e working.

**Comment by flixr** [comment]
I tested this successfully in the sandbox that @wild-endeavor so kindly created and with https://github.com/flyteorg/flytekit/pull/1515
IMHO this is good to merge.

**Review by wild-endeavor — APPROVED** [review]

**Review by eapolinario — APPROVED** [review]
